src/yetibot/commands/aws/formatters.clj

```diff
@@ -105,79 +109,122 @@
   [response]
   (let [request-id (get-in response [:UpdateLoginProfileResponse :ResponseMetadata :RequestId])]
     {:result/data  {:request-id request-id}
      :result/value (format "Login profile successfully updated [RequestId=%s]"
                            request-id)}))
 
-(defmethod format-response ::IAMAccessKeyCreated
+(defmethod format-iam-response ::IAMAccessKeyCreated
   [{{:keys [UserName AccessKeyId Status SecretAccessKey CreateDate]} :AccessKey}]
   {:result/data  {:user-name UserName, :access-key-id AccessKeyId, :status Status, :secret-access-key SecretAccessKey, :create-date CreateDate}
    :result/value (format "An access key for user %s has been successfully created on %s\nAWS Access Key ID : %s\nAWS Secret Access Key : %s\nStatus : %s\n"
                          UserName CreateDate AccessKeyId SecretAccessKey Status)})
 
-(defmethod format-response ::IAMListAccessKeysResponseReceived
+(defmethod format-iam-response ::IAMListAccessKeysResponseReceived
   [{:keys [AccessKeyMetadata]}]
   {:result/data  {:access-key-metadata AccessKeyMetadata}
    :result/value (map #(format "Access key ID : %s\nStatus : %s\nCreated on : %s\n"
                                (:AccessKeyId %) (:Status %) (:CreateDate %))
                       AccessKeyMetadata)})
 
-(defmethod format-response ::IAMAccessKeyDeleted
+(defmethod format-iam-response ::IAMAccessKeyDeleted
   [response]
   (let [request-id (get-in response [:DeleteAccessKeyResponse :ResponseMetadata :RequestId])]
     {:result/data  {:request-id request-id}
      :result/value (format "Access Key successfully deleted [RequestId=%s]"
-                           request-id)}))
\ No newline at end of file
+                           request-id)}))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defmulti format-s3-response
+          "Returns a dispatch-value matching the S3 operation that has been successfully invoked
+          or has failed"
+          (fn [response]
+            (let [aws-type (:aws/type (meta response))]
+              (cond
+                (and (s/valid? (s3-response-spec :CreateBucket) response)
+                     (= aws-type :aws.type/CreateBucket)
+                     (not (contains? response :Error))) ::S3BucketCreated
+                (and (s/valid? (s3-response-spec :ListBuckets) response)
+                     (= aws-type :aws.type/ListBuckets)
+                     (not (contains? response :Error))) ::S3BucketListed
+                (and (s/valid? (s3-response-spec :ListObjectsV2) response)
+                     (= aws-type :aws.type/ListObjects)
+                     (not (contains? response :Error))) ::S3ObjectsListed
+                :else ::error))))
+
+(defmethod format-s3-response ::error
+  [response]
+  {:result/error (get-in response [:Error :Message])})
+
+(defmethod format-s3-response ::S3BucketCreated
+  [{:keys [Location]}]
+  {:result/data  {:location Location}
```

Comment:
Are there other interesting data we could return in addition to `Location`?

Reply:
for this _bucket creation_, only the _location_ is returned if I'm not wrong. I will check that out to confirm and let you know.

Reply:
here is the actual spec for the `:CreateBucket` API call : 

```
:CreateBucket {:name "CreateBucket",
                :documentation "<p>Creates a new bucket.</p>",
                :documentationUrl "http://docs.amazonwebservices.com/AmazonS3/latest/API/RESTBucketPUT.html",
                :request {:GrantWrite string,
                          :GrantRead string,
                          :GrantReadACP string,
                          :ACL string,
                          :Bucket string,
                          :GrantWriteACP string,
                          :CreateBucketConfiguration {:LocationConstraint string},
                          :ObjectLockEnabledForBucket boolean,
                          :GrantFullControl string},
                :required [:Bucket],
                :response {:Location string}}
```

the `response` only includes the `:Location`